internal/service/ec2/vpc_ipam_preview_next_cidr.go

```diff
@@ -0,0 +1,129 @@
+package ec2
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	// "github.com/hashicorp/terraform-provider-aws/internal/flex"
+	// "github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceVPCIpamPreviewNextCidr() *schema.Resource {
+	return &schema.Resource{
+		Create: ResourceVPCIpamPreviewNextCidrCreate,
+		Read:   ResourceVPCIpamPreviewNextCidrRead,
+		Delete: schema.Noop,
+		// Having imports results in errors
+		// 	error running import: exit status 1
+		// 	Error: Error allocating cidr from IPAM pool (): InvalidParameterValue: The allocation size is too big for the pool.
+		// Importer: &schema.ResourceImporter{
+		// 	State: schema.ImportStatePassthrough,
+		// },
+		Schema: map[string]*schema.Schema{
+			"cidr": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			// // temp comment out till bug is resolved
+			// "disallowed_cidrs": {
+			// 	Type:     schema.TypeSet,
+			// 	Optional: true,
+			// 	ForceNew: true,
+			// 	Elem: &schema.Schema{
+			// 		Type: schema.TypeString,
+			// 		ValidateFunc: validation.Any(
+			// 			verify.ValidIPv4CIDRNetworkAddress,
+			// 			// Follow the numbers used for netmask_length
+			// 			validation.IsCIDRNetwork(0, 32),
+			// 		),
+			// 	},
+			// },
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipam_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"netmask_length": {
+				// Possible netmask lengths for IPv4 addresses are 0 - 32.
+				// AllocateIpamPoolCidr API
+				//   - If there is no DefaultNetmaskLength allocation rule set on the pool,
+				//   you must specify either the NetmaskLength or the CIDR.
+				//   - If the DefaultNetmaskLength allocation rule is set on the pool,
+				//   you can specify either the NetmaskLength or the CIDR and the
+				//   DefaultNetmaskLength allocation rule will be ignored.
+				// since there is no attribute to check if the rule is set, this attribute is required
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 32),
```

Comment:
we can use the const values here from the other IPAM resource: https://github.com/hashicorp/terraform-provider-aws/blob/f445f5e1980a8b4a6581b2964b8ba70b80330128/internal/service/ec2/vpc_ipam_pool_cidr_allocation.go#L56

Reply:
For this one I opted not to because the values assigned to these are specified here: https://github.com/hashicorp/terraform-provider-aws/blob/f445f5e1980a8b4a6581b2964b8ba70b80330128/internal/service/ec2/vpc.go#L24-L28

This deviates from the values specified in the API at [EC2 API reference: AllocateIpamPoolCidr](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AllocateIpamPoolCidr.html#:~:text=Required%3A%20Yes-,NetmaskLength,-The%20netmask%20length) which mentions `Possible netmask lengths for IPv4 addresses are 0 - 32. Possible netmask lengths for IPv6 addresses are 0 - 128.`

Reply:
that makes sense. its possible to allocate different values than what VPC can accept. good decision